### PR TITLE
Added mkdocs and documents deploy to the gh-pages feature

### PR DESCRIPTION
**Pull request by jpamarthi**
1. Bumped the version to 0.0.47
2. Added mkdocs feature to store the required documents and deploy them to the gh-pages
3. Automated the docs deployment to gh-pages when we merge to main branch

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -134,5 +134,6 @@ credentials.json
 .coverage
 .coverage.*
 
-# Ignore uploads directory
-uploads/
\ No newline at end of file
+# Ignore uploads and site directories
+uploads/
+site/
```
Comment (jpamarthi) [inline .gitignore:139]: This ignores the files in site directory

File: README.md [hunk 2]
```diff
@@ -5,9 +5,9 @@
 This package identifies outlier(s) for a given time-series dataset in simple steps. It supports day, week, month and 
 quarter level time-series data.
 
-- [Highlights](docs/Highlights.md)
-- [Outlier Detection Functions](docs/Outlier%20Detection%20Functions.md)
-- [Diagnostic Functions](docs/Diagnostic%20Functions.md)
+- [Highlights](docs/highlights.md)
```
Comment (jpamarthi) [inline README.md:8]: Renamed the files with snake casing for the standardization

File: mkdocs.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+site_name: PyCatcher Documentation
```
Comment (jpamarthi) [inline mkdocs.yml:1]: Mkdocs file is responsible to keep the documentation together and will deploy the config to the live site that we set up

File: pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "pycatcher"
-version = "0.0.46"
+version = "0.0.47"
```
Comment (jpamarthi) [inline pyproject.toml:3]: Bumped the version

File: pyproject.toml [hunk 5]
```diff
@@ -26,6 +26,8 @@ statsmodels = "==0.14.4"
 pyod = "==2.0.2"
 seaborn = "==0.13.2"
 flask = "^3.1.0"
+mkdocs = "^1.5.0"
```
Comment (jpamarthi) [inline pyproject.toml:29]: Added the mkdocs library for publishing the documentation

File: .github/workflows/deploy-docs.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+name: Deploy Documentation to GitHub Pages
```
Comment (jpamarthi) [inline .github/workflows/deploy-docs.yml:1]: This automates the docs deployment to gh-pages for every merge to main.

**Comment by jpamarthi** [comment]
Instructions:

1. `poetry install --with dev`
2. `poetry run mkdocs serve --verbose`
3. Open the URL on browser with address: [(http://127.0.0.1:8000/pycatcher/)]
image

**Review by aseemanand — APPROVED** [review]
LGTM

**Pull request merged**